### ISSUE TITLE
FC033: Missing template when source is not default.

### ISSUE THREAD

**Issue by rbowlby** [acrmp/foodcritic#292]
When a template source makes use of a directory under templates/ other than default it incorrectly warns. Moving the file to defaults and removing the source line corrects the issue. Should be an easy one to duplicate.

**Comment by odcinek** [comment]
Was unable to repro, @rbowlby care to provide a full example?

**Comment by philoserf** [comment]
Given a template resource

```ruby
template "/dir/file" do
  source 'file.erb'
end
```

When I run **foodcritic 4.0.0** with the template location `/template/file.erb`
Then I receive **FC033**

When I run **foodcritic 4.0.0** with the template location `/template/default/file.erb`
Then I do not receive  **FC033**

**Comment by dirkm** [comment]
I am facing the same issue, and wanted to confirm this bug-report

**Comment by soblom** [comment]
Same issue here. Foodcritic complains when template is placed directly under `template` and stops complaining when it is in a subfolder `default`. 

Foodcritic 4.0.0, installed as part of ChefDK, running on OSX

It seems we all agree that this is a false positive. There just seems to be some question on how to reproduce the bug.

**Comment by nshemonsky** [comment]
```bash
$ foodcritic -V
foodcritic 5.0.0

$ foodcritic .
FC033: Missing template: ./recipes/configure.rb:33
FC033: Missing template: ./recipes/configure.rb:49
FC033: Missing template: ./recipes/configure.rb:60
...
FC033: Missing template: ./recipes/service.rb:9

$ tree templates/
templates/
├── conf
│   ├── jboss-minimal.xml.erb
│   ├── jboss-service.xml.erb
│   ├── log4j.xml.erb
│   └── login-config.xml.erb
├── deploy
│   ├── jbossweb-tomcat55.sar
│   │   └── server.xml.erb
│   ├── jms
│   │   └── uil2-service.xml.erb
│   ├── mail-service.xml.erb
│   └── oracle-xa-ds.xml.erb
└── jboss.sh.erb
```

#### configure.rb
```ruby
template 'jboss-minimal.xml' do
  source ::File.join('conf', jboss-minimal.xml.erb)
  ...
end

template 'jboss-service.xml' do
  source 'conf/jboss-service.xml.erb'
  ...
end

...
```
Given the following layout as an example:

#### service.rb
```ruby

template '/etc/init.d/jboss' do
  source 'jboss.sh.erb'
  ...
end
```

Seeing false positives unless the source attribute paths are constructed using `::File.join`.

Additionally, the jboss.sh.erb template directly within the templates folder is also shown as a false positive but like in configure.rb, using `::File.join('jboss.sh.erb')` removes the FC033 trigger.

**Comment by joegoggins** [comment]
Seeing this as well on:

```
$ chef --version
Chef Development Kit Version: 0.17.17
chef-client version: 12.13.37
delivery version: master (f68e5c5804cd7d8a76c69b926fbb261e1070751b)
berks version: 4.3.5
kitchen version: 1.11.1
```

**Comment by troyready** [comment]
Last message was a mistake -- further tracking is in #449

**Comment by haidangwa** [comment]
I'm seeing this in foodcritic v10.0.0 and also with v9.0.0 that comes embedded with chefdk 1.2.22.

**Comment by brentm5** [comment]
I am also seeing it in chefdk version 1.3.32
```
chef -v
Chef Development Kit Version: 1.3.32
chef-client version: 12.19.36
delivery version: master (dae9922dbf96163302a1204bd219d70eb62d1e0c)
berks version: 5.6.4
kitchen version: 1.16.0

foodcritic -V
foodcritic 10.0.0
```

**Comment by tas50** [comment]
@brentm5 That's several versions of Foodcritic back which still has this issue. It'll be fixed when Foodcritic 2.0 comes out
